File: src/main/java/org/kd/main/client/presenter/PresenterHandler.java
package org.kd.main.client.presenter;

import org.kd.main.common.entities.Bank;
import org.kd.main.common.entities.Customer;
import org.kd.main.common.entities.Transfer;

import java.util.List;

public interface PresenterHandler {

    void createBank(String name, String shortname);

    Bank readBank(Long id);

    List<Bank> readBanks();

    void updateBank(Bank bank);

    void deleteBank(Long id);

    void createCustomer(String name, String shortname, Double units, Long bankId);

    Customer readCustomer(Long id);

    List<Customer> readCustomers();

    void updateCustomer(Customer customer);

    void deleteCustomer(Long id);

    void bookTransfer();

    List<Transfer> readTransfers();

    void deleteTransfer(Long id);
    
    void initApplication();

    void saveDb();

    void stopServer();
}

File: C#UI/Banque/Client/Presenter/PresenterHandler.cs
using System;
using System.Collections.Generic;
using Banque.Common.Entities;

namespace Banque.Client.Presenter
{
    public interface PresenterHandler
    {
        Boolean CreateBank(String name, String shortname);

        List<Bank> readBanks();

        Boolean updateBank(Bank bank);

        Boolean deleteBank(long id);

        Boolean createAccount(String name, String shortname, Double units, long bankId);

        List<Account> readAccounts();

        Boolean updateAccount(Account account);

        Boolean deleteAccount(long id);

        void initApplication();

        Boolean saveDb();

        void stopServer();

        void setAccountId(long accountId);

        long getAccountId();
    }
}
